### PR TITLE
graphite2: update to 1.3.12

### PR DESCRIPTION
**Pull request by dgilman**
* Updated to portgroup github
* Removed patch-src-CMakeLists.txt.diff as this is upstream
* Removed patch-src-Face.cpp.diff as it builds without this
* Re-enabled the featuremap test as it works now
* Added a test dependency on fonttools to get tests to pass
* Removed test.env as tests pass without it
* Removed "upgrade failure" cppflags-replace. I was able to do an upgrade
  successfully without it so I guess it is no longer necessary, but I am
  not super confident about what this did or why it is no longer needed.
* Enabled out_of_source build

I forced building against libc++ as shown in this ticket and it worked,
so I believe this is fixed:
Closes: https://trac.macports.org/ticket/37947

Closes: https://trac.macports.org/ticket/54141

###### Type(s)
- [x] enhancement
- [x] security fix

###### Tested on
macOS 10.13.6 17G65
Xcode 9.4.1 9F2000

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [x] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: graphics/graphite2/Portfile [hunk 1]
```diff
@@ -17,15 +18,13 @@ long_description    Graphite is a “smart font” system developed specifically
...
 
-checksums           rmd160  cc69347fcc1a7cb2b332377e51d652ccac5d712c \
-                    sha256  ec0185b663059553fd46e8c4a4f0dede60a02f13a7a1fefc2ce70332ea814567
+checksums           rmd160  f5d082e3a3c1dd5fbf18b5ef32044222875171f9 \
+                    sha256  4739d16bef8e98b96d67f79003de492deff1c6924df58a30ff1902e2a3c6c45c \
+                    size    6667476
 
-patchfiles          patch-src-CMakeLists.txt.diff \
-                    patch-src-Face.cpp.diff
+cmake.out_of_source yes
```
Comment (mf2k) [inline graphics/graphite2/Portfile]: Is there a reason not to just use the cmake 1.1 portgroup instead where this is the default?

File: graphics/graphite2/Portfile [hunk 2]
```diff
@@ -1,11 +1,12 @@
...
+PortGroup           cmake 1.1
 PortGroup           muniversal 1.0
+PortGroup           github 1.0
 
+github.setup        silnrsi graphite 1.3.12
 name                graphite2
-version             1.3.9
 categories          graphics
 platforms           darwin
 maintainers         ryandesign openmaintainer
```
Comment (pmetzger) [inline graphics/graphite2/Portfile]: This should probably be fixed to

`maintainers         {@ryandesign ryandesign} openmaintainer`

**Comment by ryandesign** [comment]
Thanks. I will look at this more tomorrow.

Since this project offers release downloads, we should use them, not the automatically-generated downloads.

> Removed "upgrade failure" cppflags-replace. I was able to do an upgrade
> successfully without it so I guess it is no longer necessary, but I am
> not super confident about what this did or why it is no longer needed.

At the time, the project's build system probably used our `CPPFLAGS` before its own, causing our `-I/opt/local/include` to precede their own build directories, causing a build failure if some previous version was installed. Changing our `-I` to `-isystem` makes our include directory get searched last, even if it is listed first on the comand line. If they've fixed the build system to put our `CPPFLAGS` after theirs, then this can be removed; if they haven't, then it can't.

**Comment by dgilman** [comment]
Where are the upstream tarballs? Taking a look at https://scripts.sil.org/cms/scripts/page.php?site_id=projects&item_id=graphite_download#aa47e0db and not seeing anything there. They seem to be pointing everyone back to Github.

**Comment by dgilman** [comment]
Here's what cmake did to build an object file, picked from the output at random:

```
[ 76%] Building CXX object src/CMakeFiles/graphite2.dir/FileFace.cpp.o
cd /opt/local/var/macports/build/_Users_david_macports-ports-local_graphics_graphite2/graphite2/work/build/src && /usr/bin/clang++  -DGRAPHITE2_NTRACING -Dgraphite2_EXPORTS -I/opt/local/var/macports/build/_Users_david_macports-ports-local_graphics_graphite2/graphite2/work/graphite-1.3.12/include -I/opt/local/var/macports/build/_Users_david_macports-ports-local_graphics_graphite2/graphite2/work/graphite-1.3.12/src  -pipe -Os -stdlib=libc++ -DNDEBUG -arch x86_64 -mmacosx-version-min=10.13 -fPIC   -Wall -Wextra -Wno-unknown-pragmas -Wimplicit-fallthrough -Wendif-labels -Wshadow -Wno-ctor-dtor-privacy -Wno-non-virtual-dtor -fno-rtti -fno-exceptions -fvisibility=hidden -fvisibility-inlines-hidden -mfpmath=sse -msse2 -std=gnu++11 -o CMakeFiles/graphite2.dir/FileFace.cpp.o -c /opt/local/var/macports/build/_Users_david_macports-ports-local_graphics_graphite2/graphite2/work/graphite-1.3.12/src/FileFace.cpp
```

I'm not seeing `-I{prefix}/include` in there at all, so I'm thinking that's why upgrades are working without the `CPPFLAGS` patch. I had a chance to do a local revision increase followed by a `port clean --all graphite2 && port upgrade graphite2` so I can confirm that the port is upgrading correctly in the face of an upgrade.

So it is your call if you want the `CPPFLAGS` patch kept in there, but it seems to me like it is no longer necessary.

**Comment by ryandesign** [comment]
> Where are the upstream tarballs? Taking a look at https://scripts.sil.org/cms/scripts/page.php?site_id=projects&item_id=graphite_download#aa47e0db and not seeing anything there. They seem to be pointing everyone back to Github.

The upstream [release files are on GitHub](https://github.com/silnrsi/graphite/releases/latest), but the Portfile, as submitted, doesn't use them; it uses automatically-generated GitHub tarballs instead, which we only want to do if upstream does not offer release files. To use the release files, add `github.tarball_from releases`, add `distname ${name}-${version}` and reinstate the removed `extract.suffix .tgz` to match the name the developer chose for the files, and recalculate the checksums.

**Comment by ryandesign** [comment]
> So it is your call if you want the `CPPFLAGS` patch kept in there, but it seems to me like it is no longer necessary.

I agree; it looks like it wasn't a graphite2 bug but rather a [bug in our cmake-1.0 portgroup](https://trac.macports.org/ticket/42872) that was fixed.

**Comment by ryandesign** [comment]
> * Add a test dependency on fonttools to get tests to pass
> * Removed `test.env` as tests pass without it

I agree that as written the `test.env` no longer makes sense, because the location it specifies for the libraries was correct for the cmake-1.0 portgroup which did an in-source build by default but is not correct for the cmake-1.1 portgroup which does an out-of-source build by default.

However, 8 of the tests fail for me on High Sierra, regardless of whether I have fonttools installed or whether I set `DYLD_LIBRARY_PATH`. What failure did you see before you added the fonttools dependency? Were you able to find any logs of what the test failure was? I can't seem to find them.

**Comment by ryandesign** [comment]
> I see `/opt/local/bin/python` in there, maybe my test successes have to do with me using `port select` to set that symlink to MacPorts Python 2.7?

Yes; I don't have python selected.

It looks like adding `configure.args-append -DPYTHON_EXECUTABLE=${prefix}/bin/python2.7` works, even if python27 is not active during the build; looks like it's only used for the tests.

**Comment by ryandesign** [comment]
Thanks for making the changes and sorry for the delay in getting around to this PR.

I've pushed a few more changes to fix the last few things. Let me check if it works on an older system before I merge it.

I see you made these changes on your fork's master, so I'm not sure exactly what will happen once we merge the PR or how you should proceed. Usually you should make a new branch before you commit changes and submit a PR, that way you can just delete the branch after the PR is merged, but since you used master that might not be the right thing to do.

**Comment by dgilman** [comment]
LGTM

**Comment by dgilman** [comment]
The Portfile has tests enabled it in but the CI didn't run them. In this case getting to run the tests on a clean system would have caught this error. Is it worth asking the CI maintainers to add running tests to the build system?

**Comment by ryandesign** [comment]
That sounds like a topic worth discussing on the macports-dev mailing list. My suspicion is that we should not enable port tests on Travis CI. A couple reasons occur to me right off the bat. 1, running tests takes time, and our builds already time out sometimes because of the limited time Travis gives us to run each build. 2, many ports' test suites fail—should this cause Travis to mark the build as failed? Is it port maintainers' responsibility to ensure upstream test suites pass?

**Comment by ryandesign** [comment]
I neglected to notice that fonttools has been updated recently to use python 3.6 instead of 2.7, so I fixed that here. The tests still pass on my High Sierra Mac. Some tests fail on my PowerPC Mac but that needn't prevent merging this commit.

> * Removed patch-src-CMakeLists.txt.diff as this is upstream

I reinstated and updated this patchfile because both of the fixes it contained were still needed (one to build on PowerPC; the other to build with GCC)—and a third fix was needed for PowerPC as well. I need to submit those fixes to the developers as a pull request, since they haven't acted on the bug reports for years.

I removed the removal of `-fno-stack-protector` on Darwin 8 because [upstream has removed it](https://github.com/silnrsi/graphite/commit/b23ae688f508050a2fb12e6443ce60783b28054e).

I removed the disabling of the segment caching code on PowerPC because [upstream has removed it](https://github.com/silnrsi/graphite/commit/b0f77e4a9dc50a888f74e904000a2486b2fc5527).

I'm keeping the old `homepage` URL because it's shorter and seems less likely to change in the future.

**Pull request merged**